### PR TITLE
Assorted macOS build fixes.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

- Use native zlib instead of a copy pulled from Homebrew.
- Use native libuuid instead of a copy pulled from Homebrew.
- Skip checking for systemd and libmnl on macOS, they’ll never be present there, so there’s no point in wasting time on them and polluting logs with the failures to find them.

##### Test Plan

CI passes on this PR, and the build works correctly on systems affected by the build failures.

### CODE REVIEW

File: packaging/cmake/Modules/NetdataDetectSystemd.cmake [hunk 1]
```diff
@@ -4,39 +4,43 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 macro(detect_systemd)
-    find_library(SYSTEMD_LIBRARY NAMES systemd)
+    if(LINUX)
```
Comment (vkalintiris) [inline packaging/cmake/Modules/NetdataDetectSystemd.cmake:7]: Don't add the OS logic within detect_systemd, instead call detect_systemd only on Linux.  No need to `set(HAVE_SYSTEMD False)` as this is the default for unset variables.

**Comment by Ferroin** [comment]
Rebased to pick up latest changes and clean up commit history.

**Review by vkalintiris — APPROVED** [review]
Is this supposed to still be in draft mode? It LGTM.

**Comment by Ferroin** [comment]
This does not completely fix the issues we have been seeing with the build on macOS, but it does improve the situation significantly, I’m merging this and will create a new PR for further work on the macOS build.

**Pull request merged**